src/main/java/use_case/graph/GraphInteractor.java

```diff
@@ -1,20 +1,38 @@
 package use_case.graph;
 
+import entity.Journey;
+import entity.Node;
+import entity.WebPage;
+import entity.WikiHistory;
+import interface_adapter.graph.GraphPresenter;
+import use_case.journey.JourneyDataAccessInterface;
+
 /**
  * The Graph Interactor.
  */
 public class GraphInteractor implements GraphInputBoundary {
     private final GraphOutputBoundary graphPresenter;
     private final GraphDataAccessInterface graphDataAccessInterface;
+    private final JourneyDataAccessInterface journeyDataAccessInterface;
 
-    public GraphInteractor(GraphOutputBoundary graphPresenter, GraphDataAccessInterface graphDataAccessInterface) {
+    public GraphInteractor(GraphOutputBoundary graphPresenter, GraphDataAccessInterface graphDataAccessInterface,
+                           JourneyDataAccessInterface journeyDataAccessInterface) {
         this.graphPresenter = graphPresenter;
         this.graphDataAccessInterface = graphDataAccessInterface;
+        this.journeyDataAccessInterface = journeyDataAccessInterface;
     }
 
     @Override
     public void execute(GraphInputData inputData) {
-        // TODO change Journey's currentNode to the one with the inputted title
+        Journey journey = journeyDataAccessInterface.getJourney();
+        WikiHistory wikiHistory = journeyDataAccessInterface.getWikiHistory();
+
+        Node node = wikiHistory.getNode(inputData.getSubject());
```

Comment:
I would change this to use only the graphDataAccessInterface methods but I'm not actually sure whether that actually better represents CA